File: tests/fixtures/sample-project/src/routes/wallet.ts
import { WalletService } from '../services/wallet';

const walletService = new WalletService();

export async function handleCreateWallet(req: Request): Promise<Response> {
  const { userId } = req.body;
  const wallet = await walletService.createWallet(userId);
  return { status: 200, body: wallet };
}

export async function handleGetBalance(req: Request): Promise<Response> {
  const { walletId } = req.params;
  const balance = await walletService.getBalance(walletId);
  return { status: 200, body: { balance } };
}

interface Request {
  body: any;
  params: any;
}

interface Response {
  status: number;
  body: any;
}

File: tests/fixtures/sample-project/src/services/wallet.go
package services

import (
	"fmt"
	"myapp/utils"
)

type Wallet struct {
	ID      string
	UserID  string
	Balance float64
}

type WalletService struct {
	logger *utils.Logger
}

// NewWalletService creates a new wallet service
func NewWalletService() *WalletService {
	return &WalletService{logger: utils.NewLogger()}
}

func (s *WalletService) CreateWallet(userID string) (*Wallet, error) {
	s.logger.Info("Creating wallet")
	utils.ValidateAddress(userID)
	return s.persistWallet(userID)
}

func (s *WalletService) GetBalance(walletID string) (float64, error) {
	s.logger.Info("Getting balance")
	return s.fetchBalance(walletID)
}

func (s *WalletService) persistWallet(userID string) (*Wallet, error) {
	return &Wallet{ID: "1", UserID: userID, Balance: 0}, nil
}

func (s *WalletService) fetchBalance(walletID string) (float64, error) {
	fmt.Println("fetching")
	return 100.0, nil
}
